staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,297 @@
+	// +optional
+	Spec PriorityLevelConfigurationSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	// `status` is the current status of a "request-priority".
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status
+	// +optional
+	Status PriorityLevelConfigurationStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityLevelConfigurationList is a list of PriorityLevelConfiguration objects.
+type PriorityLevelConfigurationList struct {
+	metav1.TypeMeta `json:",inline"`
+	// `metadata` is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// `items` is a list of request-priorities.
+	Items []PriorityLevelConfiguration `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// PriorityLevelConfigurationSpec is specification of a priority level
+type PriorityLevelConfigurationSpec struct {
+	// `exempt` defines whether the priority level is exempted or not.  There should be at most one exempt priority level.
+	// Being exempt means that requests of that priority are not subject to concurrency limits (and thus are never queued)
+	// and do not detract from the concurrency available for non-exempt requests. In a more sophisticated system, the
+	// exempt priority level would be the highest priority level. The field is default to false and only those system
+	// preset priority level can be exempt.
+	// +optional
+	Exempt bool `json:"exempt,omitempty" protobuf:"varint,1,opt,name=exempt"`
+	// `assuredConcurrencyShares` is a positive number for a non-exempt priority level, representing the weight by which
+	// the priority level shares the concurrency from the global limit. The concurrency limit of an apiserver is divided
+	// among the non-exempt priority levels in proportion to their assured concurrency shares. Basically this produces
+	// the assured concurrency value (ACV) for each priority level:
+	//
+	//             ACV(l) = ceil( SCL * ACS(l) / ( sum[priority levels k] ACS(k) ) )
+	//
+	// +optional
+	AssuredConcurrencyShares int32 `json:"assuredConcurrencyShares,omitempty" protobuf:"varint,2,opt,name=assuredConcurrencyShares"`
+	// `queues` is a number of queues that belong to a non-exempt PriorityLevelConfiguration object. The queues exist
+	// independently at each apiserver. The value must be positive for a non-exempt priority level and setting it to 1
+	// disables shufflesharding and makes the distinguisher method irrelevant.
+	// +optional
+	Queues int32 `json:"queues,omitempty" protobuf:"varint,3,opt,name=queues"`
+	// `handSize` is a small positive number for applying shuffle sharding. When a request arrives at an apiserver the
```

Comment:
Define small. (1e6 is a pretty small number compared to all the others...)

Reply:
Ah, there is a condition that is checked in validation but missing from the comments here.  The condition is that HandSize*Log_base_2(Queues) <= 60.  That should be documented, and will quantify "small".

Reply:
👍